### DISCUSSION TITLE
blog/2021/11/10/ame-bayes-re-guide/index

### DISCUSSION THREAD

**Discussion by giscus** [andrewheiss/ath-quarto#74] [Blog comments]
# blog/2021/11/10/ame-bayes-re-guide/index

How to calculate grand means, conditional group means, and hypothetical group means of posterior predictions from multilevel brms models.

https://www.andrewheiss.com/blog/2021/11/10/ame-bayes-re-guide/index.html

**Comment by GStechschulte** [comment]
Hey Andrew, fantastic post as usual. However, one part I think is contradictory is the sytnax for conditional effects in brms.

In this blog, in the section "Conditional effects for existing groups"  it is stated "to calculate this we need to...include `re_formula=NULL`. That feels wierd since `re_formula=NA` made it so there were no group effects.". 

However, in the [Marginal and conditional effects for GLMMs with marginaleffects]( https://www.andrewheiss.com/blog/2022/11/29/conditional-marginal-marginaleffects/#summary) it is stated the syntax for conditional effects in brms is `re_formula=NA` which makes it so no group effects are included.

I think this is a mix-up on the parameters passed to the `re_formula` argument.

**Reply by GStechschulte** [reply]
link to [Conditional effects for existing groups](https://www.andrewheiss.com/blog/2021/11/10/ame-bayes-re-guide/) section

**Reply by GStechschulte** [reply]
Ahh. Nevermind. I got mixed up myself. Disregard these comments. Cheers!

**Comment by courtneygoodridge** [comment]
Hi Andrew, thank you for this fantastic post. I have quick question that you do touch upon in your [Marginal and conditional effects for GLMMs with {marginaleffects}](https://www.andrewheiss.com/blog/2022/11/29/conditional-marginal-marginaleffects/#summary) post but I just wanted to get some clarification. 

In the [Conditional effects for existing groups](https://www.andrewheiss.com/blog/2021/11/10/ame-bayes-re-guide/#conditional-effects-for-existing-groups) section of this post, you suggest that setting `re_formula = NULL`  allows us to incorporate random effects into the predictions. Would this not make them *marginal effects*? In [this post](https://www.andrewheiss.com/blog/2022/11/29/conditional-marginal-marginaleffects/#conditional-effects-or-effect-of-a-variable-in-an-average-cluster) you state that *"Marginal effects = global/population-level effect; clusters on average; random offsets are incorporated into the estimate"*.  

My only thought is that when only generating a single mean (whether it includes random effects or ignores them), it still counts as conditional? *Marginal effects* are therefore expected values for specific members of the random effects structure (akin to -> [Average marginal effects](https://www.andrewheiss.com/blog/2021/11/10/ame-bayes-re-guide/#average-marginal-effects)). Am I understanding the distinction right?

**Comment by phit0** [comment]
Thanks for this gem Andrew! Just a hint for the relentlessly curious reading this -- if you want to know what is happening under the hood in emmeans and brms i found the core calculations are done in `brms:::get_new_rdraws`. So running  `debug(brms:::get_new_rdraws)` before running the code from this post might be interesting.
